src/transforms/pipelines/mod.rs

```diff
@@ -233,26 +268,52 @@ mod tests {
         outer
             .expand(name, &parents, &mut transforms, &mut expansions)
             .unwrap();
+        let routes = transforms
+            .iter()
+            .map(|(key, transform)| (key.to_string(), transform.inputs.clone()))
+            .collect::<IndexMap<String, Vec<String>>>();
+        let expansions: IndexMap<String, Vec<String>> = expansions
+            .into_iter()
+            .map(|(key, others)| {
+                (
+                    key.to_string(),
+                    others.iter().map(ToString::to_string).collect(),
+                )
+            })
+            .collect();
         assert_eq!(
             transforms
                 .keys()
                 .map(|key| key.to_string())
                 .collect::<Vec<String>>(),
             vec![
-                "foo.logs.filter",
-                "foo.logs.pipelines.foo.truthy.filter",
-                "foo.logs.pipelines.foo.truthy.transforms.0",
-                "foo.logs.pipelines.foo.truthy.transforms.1",
-                "foo.logs.pipelines.foo.truthy.transforms",
-                "foo.logs.pipelines.foo.truthy",
-                "foo.logs.pipelines.foo.falsy",
-                "foo.logs.pipelines.foo",
-                "foo.logs.pipelines.bar.0",
-                "foo.logs.pipelines.bar",
-                "foo.logs.pipelines",
-                "foo.logs",
-                "foo"
```

Comment:
Very nice! The expansion process is a bit complicated and it's difficult to see exactly how big the reduction in components is, but this looks great.